File: frontend/src/types/resource.ts
export type MatchLevel = "high" | "medium" | "low";

export interface CriterionScore {
  criterion_id: number;
  score: number;
  match_level: MatchLevel;
  reasoning: string | null;
}

export interface VideoTimestamp {
  start_seconds: number;
  end_seconds: number | null;
  label: string;
  relevance_score: number;
}

export interface YoutubeResource {
  id: number;
  node_id: number;
  video_id: string;
  title: string;
  channel_name: string | null;
  duration_seconds: number | null;
  thumbnail_url: string | null;
  youtube_url: string;
  overall_score: number;
  sort_rank: number;
  scores: CriterionScore[];
  timestamps: VideoTimestamp[];
}

File: backend/api/schemas/resource.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class CriterionScoreResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    criterion_id: int
    score: float
    match_level: str
    reasoning: Optional[str]


class TimestampResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    start_seconds: int
    end_seconds: Optional[int]
    label: str
    relevance_score: float


class YoutubeResourceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    node_id: int
    video_id: str
    playlist_id: Optional[str]
    title: str
    channel_name: Optional[str]
    duration_seconds: Optional[int]
    thumbnail_url: Optional[str]
    youtube_url: str
    overall_score: float
    sort_rank: int
    is_playlist: bool


class ResourceDetailResponse(YoutubeResourceResponse):
    scores: List[CriterionScoreResponse] = []
    timestamps: List[TimestampResponse] = []
